### PR TITLE
YohannHommet: Adding roles suggestions

### PR DESCRIPTION
**Pull request by YohannHommet**
Hi ben,
As an oclock student, we are currenbtly working with altorouter and your alto dispatcher. We tried to add roles to routes and i was wondering if we can do this using your tool. So i made few changes to add this possibility. It's pretty basic but it gives possibility to handle roles i a better way.
Hope you like it !
Thanks

### CODE REVIEW

File: Dispatcher.php [hunk 1]
```diff
@@ -63,6 +68,12 @@ public function parseTarget($target)
             if (!empty($target['controller']) && !empty($target['method'])) {
                 $this->controller = $target['controller'];
                 $this->method = $target['method'];
+
+                // Adding a key 'role' in target params to specify needed roles for current route (can be null)
+                if (isset($target['role'])) {
```
Comment (YohannHommet) [inline Dispatcher.php:73]: I would add this to handle errors if role is not specify
```php
      if (isset($target['role'])) {
          $this->role     = $target['role'];
      } else {
          $this->role = null;
      }
   ```

File: Suggestions.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    [
+        'controller' => 'MainController',
+        'method' => 'home',
+        // Set an array of roles for that route (can be set to null)
+        'role' => ['admin', 'catalog-manager']
+    ],
+    'main-home'
+);
+// Can be set like this too
+$router->map('GET', '/other-page', 'MainController::otherPage::['admin', 'user']', 'other-page');
```
Comment (YohannHommet) [inline Suggestions.md:26]: Get some errors this way :(

File: Dispatcher.php [hunk 3]
```diff
@@ -76,8 +87,11 @@ public function parseTarget($target)
                 if (strpos($target, $currentSeparator) !== false) {
                     $separatorFound = true;
                     $explodedInfos = explode($currentSeparator, $target);
+
                     $this->controller = $explodedInfos[0];
                     $this->method = $explodedInfos[1];
+                    // Not sure this can be done
+                    $this->role = $explodedInfos[2];
```
Comment (benoclock) [inline Dispatcher.php:94]: what if developer provides only two elements (controller & method) ?
and is this code compatible with current AltoRouter & AltoDispatcher use ? I think not.

**Comment by benoclock** [comment]
Your PR is interesting.
But, it means the dispatcher should not only dispatch, but also carry some data for outside purpose.

I guess you know [SRP](https://en.wikipedia.org/wiki/Single-responsibility_principle).
So, I suggest you to create a new package for the ACL purpose, related to AltoRouter and AltoDispatcher.

For now, I can't validate this PR.

File: Dispatcher.php (hunk 3 above)
Comment (YohannHommet) [reply Dispatcher.php:94]: Like this its not working perfectly, but i tried.

**Comment by YohannHommet** [comment]
Thanks for the review, i know its not perfect but i tried. The idea was to use $match in SetControllerArguments method to send needed data, i need to work on this.. Thanks a lot.

**Pull request closed without merging**